app/src/main/java/org/xbmc/kore/ui/sections/addon/AddonListFragment.java

```diff
@@ -156,13 +211,14 @@ public void onSuccess(List<AddonType.Details> result) {
 
                     adapter.clear();
                     for (AddonType.Details addon : result) {
-                        if (addon.type.equals(AddonType.Types.UNKNOWN) ||
-                            addon.type.equals(AddonType.Types.XBMC_PYTHON_PLUGINSOURCE) ||
-                            addon.type.equals(AddonType.Types.XBMC_PYTHON_SCRIPT) ||
-                            addon.type.equals(AddonType.Types.XBMC_ADDON_AUDIO) ||
-                            addon.type.equals(AddonType.Types.XBMC_ADDON_EXECUTABLE) ||
-                            addon.type.equals(AddonType.Types.XBMC_ADDON_VIDEO) ||
-                            addon.type.equals(AddonType.Types.XBMC_ADDON_IMAGE)) {
+                        if ((addon.type.equals(AddonType.Types.UNKNOWN) ||
```

Comment:
I know this is not your code. But I really think this code smells. As you touched it :). Can you make it smell less?. For instance refactoring it, by replacing the conditions with a method call `isAddonSupported(addon)`?